src/types/custom-interfaces.ts

```diff
@@ -0,0 +1,51 @@
+import { AnyToken, ERC721Token } from './signable-token';
+import { FeeEntry, GetSignableTransferRequestV1 } from '../api';
+
+// These custom interfaces are used because API.SignableToken is not yet a union type due to OAS 2.0 spec not supporting `OneOf`
+// As well as the fact that SignableToken has an extra `data` object which we would like to flatten
+// SignableToken is replaced with AnyToken
+
+export interface UnsignedOrderRequest {
+  buy: {
+    /**
+     * Fee-exclusive amount to buy the asset
+     * @type {string}
+     */
+    amount: string;
+    token: AnyToken;
+  };
+  sell: {
+    /**
+     * Amount to sell (quantity)
+     * @type {string}
+     */
+    amount: string;
+    token: AnyToken;
+  };
+  /**
+   * ExpirationTimestamp in Unix time. Note: will be rounded down to the nearest hour
+   * @type {number}
+   * @memberof UnsignedOrderRequest
+   */
+  expiration_timestamp?: number;
+  /**
+   * Inclusion of either maker or taker fees
+   * @type {Array<FeeEntry>}
+   * @memberof UnsignedOrderRequest
+   */
+  fees?: Array<FeeEntry>;
+}
+
+export interface UnsignedTransferRequest
+  extends Omit<GetSignableTransferRequestV1, 'token' | 'sender'> {
+  token: AnyToken;
+}
+
+export interface NftTransferDetails extends Omit<ERC721Token, 'type'> {
```

Comment:
Same comment here, let's just type out the whole interface

Reply:
[502c68a](https://github.com/immutable/imx-core-sdk/pull/149/commits/502c68a0ab770b0463b90a249b37de5ae524fb77)